### PR TITLE
Pick the most parallelized input as the reference.

### PR DESCRIPTION
**Pull request by wujingyue**
See code comments for details. 

This is required to land #3045.

### CODE REVIEW

File: csrc/preseg_passes/propagate_shardings.cpp [hunk 1]
```diff
@@ -16,29 +16,8 @@
 
 namespace nvfuser::preseg_passes {
 
-void PropagateShardingsPass::runPass(Fusion* fusion) {
```
[the remark this answers is not in the document]
Comment (wujingyue) [reply csrc/preseg_passes/propagate_shardings.cpp]: Never mind -- I fixed this in an update.

**Comment by wujingyue** [comment]
!build

**Review by cowanmeg — APPROVED** [review]
Nice!

**Comment by wujingyue** [comment]
!build

**Comment by wujingyue** [comment]
!build

**Comment by wujingyue** [comment]
!build

**Comment by wujingyue** [comment]
!build

**Pull request merged**